ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableIQv2.java

```diff
@@ -63,23 +63,13 @@
       final StateQueryResult<ValueAndTimestamp<GenericRow>>
           result = stateStore.getKafkaStreams().query(request);
 
-      final Map<Integer, QueryResult<ValueAndTimestamp<GenericRow>>> queryResultMap =
-          result.getPartitionResults();
-      if (queryResultMap.size() != 1) {
-        throw new MaterializationException("The query did not return exactly one partition result: "
-                + result.getPartitionResults());
-      }
-      if (!queryResultMap.containsKey(partition)) {
-        throw new MaterializationException(
-            "The query did not return the requested partition result: "
-                + result.getPartitionResults());
-      }
       final QueryResult<ValueAndTimestamp<GenericRow>> queryResult =
-          queryResultMap.get(partition);
+          result.getPartitionResults().get(partition);
+      // Some of these failures are retriable, and in the future, we may want to retry
+      // locally before throwing.
       if (queryResult.isFailure()) {
-        throw new MaterializationException(queryResult.getFailureMessage());
-      }
-      if (queryResult.getResult() == null) {
+        throw failedQueryException(queryResult);
```

Comment:
I initially suggested just calling `getResult` and letting the FailedResult class throw its own exception, but decided that's inappropriate, since that exception's message implies you shouldn't attempt to get the result of a failed query.

In any case, I thought we could give a little more context in the exception message, so I expanded on the message we throw.